llvm/lib/Target/AIE/AIEMachineScheduler.cpp

```diff
@@ -1146,8 +1153,9 @@ void AIEScheduleDAGMI::recordDbgInstrs(const Region &CurrentRegion) {
   // We connect any Debug machine instruction to the instruction before it.
   // if there is no instruction before it, it is recorded in FirstDbgValue;
   MachineInstr *DbgMI = nullptr;
-  for (auto MII = CurrentRegion.end(), MIE = CurrentRegion.begin(); MII != MIE;
-       --MII) {
+  for (auto MII = CurrentRegion.free_instrs().end(),
+            MIE = CurrentRegion.free_instrs().begin();
+       MII != MIE; --MII) {
```

Comment:
nit: for over reverse range?

Reply:
Yes I think I could do that in an NFC commit